R/nest.R

```diff
@@ -74,20 +112,32 @@
 #'   dplyr::group_by(fish) %>%
 #'   nest()
 #'
+#' # That is similar to `nest(.by = )`, except here the result isn't grouped
+#' fish_encounters %>%
+#'   nest(.by = fish)
+#'
 #' # Nesting is often useful for creating per group models
 #' mtcars %>%
-#'   dplyr::group_by(cyl) %>%
-#'   nest() %>%
+#'   nest(.by = cyl) %>%
 #'   dplyr::mutate(models = lapply(data, function(df) lm(mpg ~ wt, data = df)))
-nest <- function(.data, ..., .names_sep = NULL, .key = deprecated()) {
-  cols <- enquos(...)
+nest <- function(.data,
+                 ...,
+                 .by = NULL,
+                 .key = NULL,
+                 .names_sep = NULL) {
+  if (identical(.key, deprecated())) {
+    # Temporary support for S3 method authors that set `.key = deprecated()`
+    return(nest(.data, ..., .by = {{ .by }}, .key = NULL, .names_sep = .names_sep))
+  }
```

Comment:
My hack to hopefully catch S3 methods that set `.key = deprecated()`. I know tibbletime did that. It doesn't look like sf or tsibble do it, so hopefully no big issues.

Reply:
This didn't work quite right. See https://github.com/tidyverse/tidyr/pull/1461/commits/ab2673dda0047846b5829c0481cb89477d008a4a